internal/provider/resources/concurrency_limit.go

```diff
@@ -0,0 +1,263 @@
+		resp.Diagnostics.Append(helpers.ResourceClientErrorDiagnostic("Concurrency Limit", "create", err))
+
+		return
+	}
+
+	copyConcurrencyLimitToModel(concurrencyLimit, &plan)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func copyConcurrencyLimitToModel(concurrencyLimit *api.ConcurrencyLimit, model *ConcurrencyLimitResourceModel) diag.Diagnostics {
+	model.ID = types.StringValue(concurrencyLimit.ID.String())
+	model.Created = customtypes.NewTimestampValue(*concurrencyLimit.Created)
+	model.Updated = customtypes.NewTimestampValue(*concurrencyLimit.Updated)
+	model.Tag = types.StringValue(concurrencyLimit.Tag)
+	model.ConcurrencyLimit = types.Int64Value(concurrencyLimit.ConcurrencyLimit)
+
+	return nil
+}
+
+// Delete deletes the resource.
+func (r *ConcurrencyLimitResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state ConcurrencyLimitResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client, err := r.client.ConcurrencyLimits(state.AccountID.ValueUUID(), state.WorkspaceID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.Append(helpers.CreateClientErrorDiagnostic("Concurrency Limit", err))
+
+		return
+	}
+
+	err = client.Delete(ctx, state.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.Append(helpers.ResourceClientErrorDiagnostic("Concurrency Limit", "delete", err))
+
+		return
+	}
+}
+
+// Read reads the resource state from the API.
+func (r *ConcurrencyLimitResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state ConcurrencyLimitResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client, err := r.client.ConcurrencyLimits(state.AccountID.ValueUUID(), state.WorkspaceID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.Append(helpers.CreateClientErrorDiagnostic("Concurrency Limit", err))
+
+		return
+	}
+
+	concurrencyLimit, err := client.Read(ctx, state.ID.ValueString())
+	if err != nil {
+		resp.Diagnostics.Append(helpers.ResourceClientErrorDiagnostic("Concurrency Limit", "get", err))
+
+		return
+	}
+
+	copyConcurrencyLimitToModel(concurrencyLimit, &state)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update updates the resource state.
+// This resource does not support updates.
+func (r *ConcurrencyLimitResource) Update(_ context.Context, _ resource.UpdateRequest, _ *resource.UpdateResponse) {
+}
+
+// ImportState imports the resource into Terraform state.
+func (r *ConcurrencyLimitResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	// we'll allow input values in the form of:
+	// - "id,workspace_id"
+	// - "id"
+	maxInputCount := 2
+	inputParts := strings.Split(req.ID, ",")
+
```

Comment:
```suggestion

	// eg. ",foo" or "foo,"
	if len(inputParts) == maxInputCount && (inputParts[0] == "" || inputParts[1] == "") {
		resp.Diagnostics.AddError(
			"Unexpected Import Identifier",
			fmt.Sprintf("Expected non-empty import identifiers, in the form of `id,workspace_id`. Got %q", req.ID),
		)

		return
	}
```

This will catch if the user provides an empty identifier in one of the positions (we should really make this a reusable helper instead of defining it in every resource but this works for now)